vault/identity_store_injector_testonly.go

```diff
@@ -0,0 +1,836 @@
+		if !ok {
+			metadata = make(map[string]string)
+		}
+		flags := DuplicateEntityFlags{
+			CommonDuplicateFlags: CommonDuplicateFlags{
+				Name:          data.Get("name").(string),
+				NamespaceID:   data.Get("namespace_id").(string),
+				DifferentCase: data.Get("different_case").(bool),
+				Metadata:      metadata.(map[string]string),
+			},
+			Policies: data.Get("policies").([]string),
+			Count:    data.Get("count").(int),
+		}
+
+		if flags.Count < 1 {
+			flags.Count = 2
+		}
+
+		ids, err := i.CreateDuplicateEntitiesInStorage(ctx, flags)
+		if err != nil {
+			i.logger.Error("error creating duplicate entities", "error", err)
+			return logical.ErrorResponse("error creating duplicate entities"), err
+		}
+
+		return &logical.Response{
+			Data: map[string]interface{}{
+				"entity_ids": ids,
+			},
+		}, nil
+	}
+}
+
+func (i *IdentityStore) createDuplicateEntityAliases() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+		metadata, ok := data.GetOk("metadata")
+		if !ok {
+			metadata = make(map[string]string)
+		}
+
+		flags := DuplicateEntityAliasFlags{
+			CommonDuplicateFlags: CommonDuplicateFlags{
+				Name:          data.Get("name").(string),
+				NamespaceID:   data.Get("namespace_id").(string),
+				DifferentCase: data.Get("different_case").(bool),
+				Metadata:      metadata.(map[string]string),
+			},
+			CommonAliasFlags: CommonAliasFlags{
+				MountAccessor: data.Get("mount_accessor").(string),
+			},
+			Count: data.Get("count").(int),
+		}
+
+		if flags.Count < 1 {
+			flags.Count = 2
+		}
+
+		ids, err := i.CreateDuplicateEntityAliasesInStorage(ctx, flags)
+		if err != nil {
+			i.logger.Error("error creating duplicate entities", "error", err)
+			return logical.ErrorResponse("error creating duplicate entities"), err
+		}
+
+		return &logical.Response{
+			Data: map[string]interface{}{
+				"entity_ids": ids,
+			},
+		}, nil
+	}
+}
+
+func (i *IdentityStore) createDuplicateLocalEntityAlias() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+		metadata, ok := data.GetOk("metadata")
+		if !ok {
+			metadata = make(map[string]interface{})
+		}
+
+		flags := DuplicateEntityAliasFlags{
+			CommonDuplicateFlags: CommonDuplicateFlags{
+				Name:        data.Get("name").(string),
+				NamespaceID: data.Get("namespace_id").(string),
+				Metadata:    metadata.(map[string]string),
+			},
+			CommonAliasFlags: CommonAliasFlags{
+				MountAccessor: data.Get("mount_accessor").(string),
+				CanonicalID:   data.Get("canonical_id").(string),
+			},
+		}
+
+		if flags.Name == "" {
+			return logical.ErrorResponse("name is required"), nil
+		}
+		if flags.CanonicalID == "" {
+			return logical.ErrorResponse("canonical_id is required"), nil
+		}
+		if flags.MountAccessor == "" {
+			return logical.ErrorResponse("mount_accessor is required"), nil
+		}
```

Comment:
🤔 should we have similar validation for other handlers?

Reply:
maybe it's not required? I think it really depends on what's the test will try to achieve...